template/module/tests/test_abstract_something.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Copyright <YEAR(S)> <AUTHOR(S)>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, models
+from openerp.tests.common import SingleTransactionCase
+
+
+class AbstractSomethingTester(models.TransientModel):
+    """ It provides a real model object to test the abstract with """
+    _name = 'abstract.something.tester'
+    _description = 'Abstract Something Tester'
+    _inherit = 'abstract.something'
+
+
+class TestAbstractSomething(SingleTransactionCase):
+    @classmethod
+    def _init_test_model(cls, model_cls):
+        """ It builds a model from model_cls in order to test abstract models
+
+        Args:
+            model_cls: (openerp.models.BaseModel) Class of model to initialize
+        Returns:
+            Model instance
+        """
+        registry = cls.env.registry
+        cr = cls.env.cr
+        inst = model_cls._build_model(registry, cr)
+        model = cls.env[model_cls._name].with_context(todo=[])
+        model._prepare_setup()
+        model._setup_base(partial=False)
+        model._setup_fields(partial=False)
+        model._setup_complete()
+        model._auto_init()
+        model.init()
+        model._auto_end()
+        return inst
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestAbstractSomething, cls).setUpClass()
+        cls.env.registry.enter_test_mode()
+        cls.old_cursor = cls.cr
+        cls.cr = cls.registry.cursor()
+        cls.env = api.Environment(cls.cr, cls.uid, {})
+        cls._init_test_model(AbstractSomethingTester)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.env.registry.leave_test_mode()
+        cls.cr = cls.old_cursor
+        cls.env = api.Environment(cls.cr, cls.uid, {})
+        super(TestAbstractSomething, cls).tearDownClass()
```

Comment:
A damned fine question - I just had to [look this up](https://docs.python.org/2/library/functions.html#super) to find out. Super accepts an `object-or-type` as the second argument for binding. It even specifically mentions classmethods:

> If the second argument is a type, issubclass(type2, type) must be true (this is useful for classmethods).